api/machine/machine.proto

```diff
@@ -847,10 +847,13 @@ message InstallConfig {
 
 message MachineConfig {
   enum MachineType {
+    option allow_alias = true;
     TYPE_UNKNOWN = 0;
     TYPE_INIT = 1;
     TYPE_CONTROL_PLANE = 2;
-    TYPE_JOIN = 3;
+    TYPE_WORKER = 3;
+    // Alias for TYPE_WORKER.
+    TYPE_JOIN = 3 [deprecated = true];
```

Comment:
Never having used the `deprecated` option before, does that mean both `TYPE_WORKER` and `TYPE_JOIN` can both be field 3?

Reply:
In combination with `option allow_alias = true` – yes: https://github.com/AlekSi/talos/blob/9039fa6e9176d6d1a02c600eff0835a5bb146317/pkg/machinery/api/machine/machine.pb.go#L294-L298